Update instant booted sims (#1915)
saucelabs/sauce-docs: docs/mobile-apps/features/instant-booted-simulators.md

---
id: instant-booted-simulators
title: Instant Booted iOS Simulators
sidebar_label: Instant Booted iOS Simulators
---

import useBaseUrl from '@docusaurus/useBaseUrl';
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

<p><small><span className="sauceDBlue">Beta</span></small></p>

The speed of the Simulator is a crucial factor that impacts the development workflow of iOS app developers. Instant booted iOS Simulators offer significant advantages over our default Simulators, as they reduce the time it takes to launch the Simulator to test apps to almost zero seconds.

> With Instant Booted iOS Simulators, you can quickly switch between two iOS versions and test your app in four configurations to ensure the compatibility and performance of your app.

Reviewer: With Instant Booted iOS Simulators, you can quickly switch between two iOS versions and test your app in four configurations to ensure its compatibility and performance.

Author: ```suggestion
With Instant Booted iOS Simulators, you can quickly switch between two iOS versions and test your app in four configurations to ensure its compatibility and performance.
```

Author: Thanks @eugenekatsov ,

I've changed it

To ensure that your capabilities remain consistent and you always have access to an instant-booted iPhone or iPad, we introduced two aliases for the device names. These aliases are as follows:

- `iPhone Instant Simulator`
- `iPad Instant Simulator`

We also introduced iOS Version Management, see [iOS Version Management](#), which add three new platform version aliases. These are

- `latest|current_major`
- `previous_major`

The combination of the device name and platform version alias will automatically allocate an Instant Booted Simulator for your manual or automated test.

:::caution
The alias `iPhone Instant Simulator` or `iPad Instant Simulator` will only work with the platform version aliases `latest|current_major` and `previous_major`. If you use the alias `iPhone Instant Simulator` or `iPad Instant Simulator` with a specific platform version, for example `16.2`, then the test will fail with the following error message

```
Error: Failed to create session.
Invalid platform version specified for instant Simulators: iPhone Instant Simulator. Must be one of ['previous_major', 'current_major', 'latest']
```

:::

There are two advantages of using the aliases

> 1. When Apple releases a new major version of iOS they always release a new iPhone or iPad. Sauce Labs will automatically assign this model to the aliases `iPhone Instant Simulator` or `iPad Instant Simulator` so you will automatically get access to the new device without having to change your capabilities. This means that you will automatically use the latest device for your tests and will be testing your app on what will most likely be the most popular device used by your users.
> 2. When Apple releases a new minor or major iOS version and Sauce Labs adds this to their cloud, then you will automatically get access to the new version without having to change your capabilities. This means that you will automatically use the latest iOS version for your tests and will be testing your app on what is most likely the most popular iOS version used by your users.

Reviewer 2: ```suggestion
1. Whenever Apple releases a new major version of iOS, they usually launch a new iPhone or iPad model. Sauce Labs assigns these models to the aliases `iPhone Instant Simulator` or `iPad Instant Simulator`, which means you can access the new device automatically without having to change your capabilities. Consequently, you'll be using the latest device for your tests and will be testing your app on what will most likely be the most popular device used by your users.

2. If Apple releases a new minor or major iOS version and Sauce Labs incorporates it into their cloud, you'll get automatic access to the new version without having to alter your capabilities. This means that you'll be using the latest iOS version for your tests and will be testing your app on what is most likely the most popular iOS version used by your users.
```

:::info
The following examples use iOS 16 and 15 as the current and previous major versions of iOS, respectively.
:::

| Device Name              | Platform Version                                |              Result              |   Instant Booted   | Note                                                                                                                                                                        |
| :----------------------- | :---------------------------------------------- | :------------------------------: | :----------------: | :-------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | iPhone Instant Simulator | latest/current_major                            |       iPhone 14 with 16.2        | :white_check_mark: | The requested instant-booted device alias, requested OS version and latest OS version installed in our cloud match. This means an instant-booted simulator will be started. |

Author: ```suggestion
| iPhone Instant Simulator | latest/current_major                            |       iPhone 14 with 16.2        | :white_check_mark: | The requested instant-booted device alias, requested OS version, and latest OS version installed in our cloud match. This means an instant-booted Simulator will be started. |
```

| iPad Instant Simulator   | previous_major                                  | iPad (9th Generation) with 15.5  | :white_check_mark: |                                                                                                                                                                              |
| iPhone Instant Simulator | A fixed iOS version, for example, 16.2, or 15.5 |                -                 |        :x:         | The combination is invalid. The device name alias **CAN ONLY** be combined with platform version aliases                                                                     |
| iPad Instant Simulator   | A fixed iOS version, for example, 16.2, or 15.5 |                -                 |        :x:         |                                                                                                                                                                              |

### Using explicit capabilities

> You can also use explicit capabilities to access the Instant Booted iOS Simulators. This is not recommended as it is more likely to land on a normal booted Simulator when Apple releases a new minor or major version of iOS. This means that you will not automatically get access to the latest device and or iOS version for your tests and will be testing your app on what is most likely not the most popular device and or iOS version used by your users.

Reviewer 2: ```suggestion
You can use explicit capabilities to access the Instant Booted iOS Simulators. However, this is not recommended because you're more likely to end up on a normally booted simulator when Apple releases a new minor or major version of iOS. As a result, you won't automatically gain access to the latest device or iOS version for your tests, and you'll likely be testing your app on a less popular device or iOS version that is not commonly used by your users.
```

> If you want to use explicit capabilities, then you you can use the following combinations

Reviewer 2: ```suggestion
If you want to use explicit capabilities, then you you can use the following combinations:
```

> If you want to use explicit capabilities, then you you can use the following combinations:

Reviewer 2: ```suggestion
If you want to use explicit capabilities, then you can use the following combinations:
```
